app/utils/gtoken.ts

```
@@ -0,0 +1,285 @@
+    }
+
+    return this.#getTokenAsync(opts);
+  }
+
+  async #getTokenAsync(opts: GetTokenOptions): Promise<TokenData | undefined> {
+    if (this.#inFlightRequest && !opts.forceRefresh) {
+      return this.#inFlightRequest;
+    }
+
+    try {
+      return await (this.#inFlightRequest = this.#getTokenAsyncInner(opts));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)